src/ansys/dpf/post/modal_mechanical_simulation.py

```diff
@@ -0,0 +1,4053 @@
+import warnings
+
+from ansys.dpf import core
+from ansys.dpf.post.data_object import DataObject
+from ansys.dpf.post.selection import Selection
+from ansys.dpf.post.simulation import MechanicalSimulation, ResultCategory
+
+
+class ModalMechanicalSimulation(MechanicalSimulation):
+    """Provides methods for mechanical modal simulations."""
+
+    def _get_result(
+        self,
+        base_name: str,
+        location: str,
+        category: ResultCategory,
+        components: Union[str, List[str], int, List[int], None] = None,
+        norm: bool = False,
+        node_ids: Union[List[int], None] = None,
+        element_ids: Union[List[int], None] = None,
+        frequencies: Union[float, List[float], None] = None,
+        set_ids: Union[int, List[int], None] = None,
+        all_sets: bool = False,
+        load_steps: Union[
+            int, List[int], Tuple[int, Union[int, List[int]]], None
+        ] = None,
+        named_selections: Union[List[str], str, None] = None,
+        selection: Union[Selection, None] = None,
+    ) -> DataObject:
+        """Extract stress results from the simulation.
+
+        Arguments `selection`, `set_ids`, `all_sets`, `times`, and `load_steps` are mutually
+        exclusive.
+        If none of the above is given, only the last result will be returned.
+
+        Arguments `selection`, `named_selections`, `element_ids`, and `node_ids` are mutually
+        exclusive.
+        If none of the above is given, results will be extracted for the whole mesh.
+
+         Args:
+            base_name:
+                Base name for the requested result.
+            location:
+                Location requested.
+            category:
+                Type of result requested. See the :class:`ResultCategory` class.
+            components:
+                Components to get results for.
+            norm:
+                Whether to return the norm of the results.
+            selection:
+                Selection to get results for.
+                A Selection defines both spatial and time-like criteria for filtering.
+            times:
+                List of times to get results for.
+            set_ids:
+                List of sets to get results for.
+                A set is defined as a unique combination of {time, load step, sub-step}.
+            all_sets:
+                Whether to get results for all sets.
+            load_steps:
+                List of load steps to get results for.
+            node_ids:
+                List of IDs of nodes to get results for.
+            element_ids:
+                List of IDs of elements to get results for.
+            named_selections:
+                Named selection or list of named selections to get results for.
+
+        Returns
+        -------
+            Returns a :class:`ansys.dpf.post.data_object.DataObject` instance.
+
+        """
+        # Build the targeted time scoping
+        time_scoping = self._build_time_freq_scoping(
```

Comment:
@PProfizi sorry to see that late, but didn't we say we would use the selection object for that? To centralize the code implementation on it

Reply:
also be cause we need to work withh workflows and it's what the selection does

Reply:
@cbellot000 you are right, this first version was using the Selection API to create one on-the-fly and use its workflow.
I've added this for ModalMechanicalSimulation.